### PR TITLE
API-421: Get a product draft

### PR DESCRIPTION
**Pull request by damien-carcel**
~Needs https://github.com/akeneo/api-php-client-ee/pull/4 to be merged first.~ Done

### CODE REVIEW

**Review by momoss — APPROVED** [review]

File: src/Api/ProductDraftApi.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+namespace Akeneo\PimEnterprise\Api;
+
+use Akeneo\Pim\Client\ResourceClientInterface;
+use Akeneo\Pim\Pagination\PageFactoryInterface;
+use Akeneo\Pim\Pagination\ResourceCursorFactoryInterface;
+
+/**
+ * API implementation to manage product drafts.
+ *
+ * Allows to:
```
Comment (ahocquard) [inline src/Api/ProductDraftApi.php]: not useful

File: tests/Common/Api/ProductDrafts/GetProductDraftApiIntegration.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                        'medium_boot',
+                    ],
+                ],
+            ],
+            'metadata' => [
+                'workflow_status' => 'draft_in_progress',
+            ],
+        ];
+
+        $api = $this->createClient('Sandra', 'Sandra')->getProductDraftApi();
```
Comment (ahocquard) [inline tests/Common/Api/ProductDrafts/GetProductDraftApiIntegration.php]: mhh weird to allow configuration on the user, but hardfix it in this case.

File: tests/Common/Api/ProductDrafts/GetProductDraftApiIntegration.php [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        $productDraft = $api->get('big_boot');
+        $sanitizedProductDraft = $this->sanitizeProductDraftData($productDraft);
+
+        $this->assertSame($expectedProductDraft, $sanitizedProductDraft);
+    }
+
+    /**
+     * @expectedException \Akeneo\Pim\Exception\NotFoundHttpException
+     */
+    public function testCannotGetIfNoRight()
```
Comment (ahocquard) [inline tests/Common/Api/ProductDrafts/GetProductDraftApiIntegration.php]: RightAccess

File: tests/Common/Api/ApiTestCase.php [hunk 4]
```diff
@@ -15,9 +15,12 @@
 abstract class ApiTestCase extends BaseApiTestCase
 {
     /**
+     * @param string $user
+     * @param string $password
+     *
      * @return AkeneoPimEnterpriseClientInterface
      */
-    protected function createClient()
+    protected function createClient($user = '', $password = '')
```
Comment (ahocquard) [inline tests/Common/Api/ApiTestCase.php:23]: should be hardcoded with admin in another PR

**Review by ahocquard — APPROVED** [review]
